### PR TITLE
Improve Userguide examples

### PR DESCRIPTION
**Pull request by ewanv**
Simplifying the early userguide examples by changing MOVIE_RELEASE_YEAR features to USER_AGE features, removing the need for date parsing.

Moved the alternate sink example to the advanced section, as it's an advanced feature.

### CODE REVIEW

**Comment by triggerNZ** [comment]
:+1:

File: USERGUIDE.markdown [hunk 1]
```diff
@@ -138,22 +133,19 @@ For details of the other classes available, refer to the **Advanced** section.
...
-import commbank.coppersmith.util.Datestamp
 import commbank.coppersmith.api._
-import commbank.coppersmith.examples.thrift.Movie
+import commbank.coppersmith.examples.thrift.User
 
-object MovieFeatures extends BasicFeatureSet[Movie] {
+object UserFeatures extends BasicFeatureSet[User] {
   val namespace              = "userguide.examples"
-  def entity(movie: Movie)   = movie.id
+  def entity(user: User)     = user.id
```
Comment (dkristian) [inline USERGUIDE.markdown]: Minor: space before `=` above can be reduced while maintaining vertical alignment

File: USERGUIDE.markdown [hunk 2]
```diff
@@ -138,22 +133,19 @@ For details of the other classes available, refer to the **Advanced** section.
...
-    (movie) => parse(movie.releaseDate).right.toOption.map(_.year)
+  val userAge = basicFeature[Integral](
+    "USER_AGE", "Age of user in years", Continuous,
+    (user) => user.age
   )
 
-  val features = List(movieReleaseYear)
+  val features = List(userAge)
 }
 ```
```
Comment (dkristian) [inline USERGUIDE.markdown]: The text below this line reads
> Note that, as opposed to above, returning `None` as part of `basicFeature` will generate a feature with a value of `null`.

I think that should be reworded as
> Note that, as opposed to above, returning `None` from the function passed to `basicFeature` will generate a feature with a value of `null`.

File: USERGUIDE.markdown [hunk 3]
```diff
@@ -468,7 +391,7 @@ object Implicits {
     val parse = Datestamp.parseFormat("dd-MMM-yyyy")
 
     def safeReleaseDate: Option[Datestamp] =
-      parse(movie.releaseDate).right.toOption
+      movie.releaseDate.flatMap(parse(_).right.toOption)
     def releaseYear:     Option[Int]      = safeReleaseDate.map(_.year)
     def isComedy:        Boolean          = movie.comedy == 1
     def isFantasy:       Boolean          = movie.fantasy == 1
```
Comment (dkristian) [inline USERGUIDE.markdown]: Minor: Add additional space before `=` above and below to align with `safeReleaseDate`, or alternatively, create a separate group and reduce space.

File: USERGUIDE.markdown [hunk 4]
```diff
@@ -1034,19 +953,24 @@ package commbank.coppersmith.examples.userguide
 import commbank.coppersmith.api._, Coppersmith._
 import commbank.coppersmith.examples.thrift.Movie
 
+import Implicits.RichMovie
+
 object SourceViewFeatures extends FeatureSetWithTime[Movie] {
   val namespace              = "userguide.examples"
   def entity(movie: Movie)   = movie.id
```
Comment (dkristian) [inline USERGUIDE.markdown]: Minor: can reduce space before `=` above.

**Comment by dkristian** [comment]
Couple of minor formatting issues, otherwise, 👍

**Pull request merged**